src/common/cockpitlocale.c

```diff
@@ -103,6 +102,7 @@ cockpit_locale_set_language (const gchar *value)
       g_setenv ("LANG", locale, TRUE);
     }
 
-  strncpy (previous, locale, sizeof (previous));
+  strncpy (previous, value, sizeof (previous) - 1);
```

Comment:
Shouldn't we be copying the length of value and then adding the null termination.